src/types.cpp

```diff
@@ -0,0 +1,47 @@
+#include "create/types.h"
+
+namespace create {
+  RobotModel::RobotModel(const std::string& name) {
```

Comment:
Maybe I am misunderstanding you, but I'm not sure how this would work. This constructor is used to convert a string into a `RobotModel`, which is used to implement the parsing of the robot_model parameter in create_autonomy.

I was basically trying to implement similar behavior to the Java [`Enum.valueOf()`] (https://docs.oracle.com/javase/8/docs/api/java/lang/Enum.html#valueOf-java.lang.Class-java.lang.String-) method, and while I could do this with a lookup table, it seems overkill for this situation.